src/lib/comms/include/sol-http.h

```diff
@@ -238,7 +239,15 @@ bool sol_http_param_add(struct sol_http_param *params,
     struct sol_http_param_value value) SOL_ATTR_WARN_UNUSED_RESULT;
 void sol_http_param_free(struct sol_http_param *params);
 
-int sol_http_escape_string(char **escaped, const char *value);
+int sol_http_escape_string(char **escaped, const struct sol_str_slice value);
```

Comment:
I wonder if we could benefit from `sol_buffer` in these APIs.

In the case of `sol_http_escape_string()` we could do a trick similar to what we did in JSON, where we take an unintialized buffer, see if we need to escape and do it using default buffer flags and if no escape was needed, we set flags to `MEMORY_NOT_OWNED` and set buffer to `value` (data/len). At the end users should always use buffer length and then call `sol_buffer_fini()` to release memory if it was escaped.

Talk to @otaviobp to see if he think it may help.